tsconfig.json

```diff
@@ -1,42 +1,31 @@
 {
+  "extends": "@canonical/typescript-config-react",
   "compilerOptions": {
     "outDir": "./public/js/dist",
     "baseUrl": ".",
     "paths": {
       "api/*": ["src/api/*"],
       "context/*": ["src/context/*"],
       "components/*": ["src/components/*"],
+      "lib/*": ["src/lib/*"],
       "pages/*": ["src/pages/*"],
       "types/*": ["src/types/*"],
       "util/*": ["src/util/*"],
       "*": ["types/*.d.ts"]
     },
-    "target": "ES2020",
-    "module": "esnext",
-    "jsx": "react-jsx",
-    "strict": true,
-    "esModuleInterop": true,
     "skipLibCheck": true,
-    "forceConsistentCasingInFileNames": true,
-    "lib": ["dom", "dom.iterable", "esnext"],
-    "allowJs": true,
-    "allowSyntheticDefaultImports": true,
-    "noFallthroughCasesInSwitch": true,
-    "moduleResolution": "node",
```

Comment:
Can we maybe keep the `moduleResolution: node` setting so that we don't have to update all the file imports in the codebase?

Reply:
maybe an upstream suggestion?

Reply:
@mas-who Indeed we are working with ES modules for the new react component library. 

I don't see any issues with overriding the module settings in the project tsconfig. You'd also need to retain your existing `esModuleInterop` and `allowSyntheticDefaultImport` settings in this case.

```json
    "module": "esnext",
    "moduleResolution": "node",
    "esModuleInterop": true,
    "allowSyntheticDefaultImports": true,
```

CC @advl

Reply:
Yes, on 2nd thought we shouldn't mess with the imports or backporting anything will be a big pain. thx for the suggestion @mas-who  and thx for the new suggestion @jmuzina Adjusted the config and reverted the imports back to how they were before.

Reply:
On Monday I'll open up a MM thread for more discussion about what sort of module settings & import syntax you'd like to see in the future - there are benefits to using `NodeNext` (increased strictness, less ambiguity, consistency between dev & production environments), but possibly we only need some of these for the library itself, and the base config could be slightly less strict. 

Looking forward to the discussion :)

Reply:
Thanks @jmuzina :)

Reply:
@edlerd @mas-who  I've tried to summarize the module settings to get a conversation going about what is the best way forward for the configurations and consuming apps. [Have a look](https://coda.io/d/Design-System_d5tlz-HCRxz/DS-12-Develop-new-Code-Architecture_sumBJY72#DS-12-Notes_tubUgmzW/r125&view=full) and let me know what you think, thanks :)